### PR TITLE
Lab1 fibinocci: creating a new PR from a branch for Fibinocci

### PR DESCRIPTION
**Pull request by anishaNair**
Fixes #

#### Changes proposed in this PR:

-
-

### CODE REVIEW

File: 01_fibonacci/main.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	for num := 1; num < 10; num++ {
+		fmt.Fprintln(out, fibinocci(num))
+	}
+}
```
Comment (anzdaddy) [inline 01_fibonacci/main.go:15]: Add a blank line between functions.

File: 01_fibonacci/main.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+func fibinocci(x int) int {
+	if x <= 0 {
+		return 0
+	} else if x < 2 {
```
Comment (anzdaddy) [inline 01_fibonacci/main.go:19]: Idiomatic Go omits the `else` when the `if` has a `return`:
```go
if x <= 0 {
	return 0
}
if x < 2 {
	return x
}
```
A more pertinent issue is that fibonacci sequences aren't defined for negative numbers, so you either should fail the call or leave it undefined (letting the `x < 2` clause handle them).

File: 01_fibonacci/main.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	for num := 1; num < 10; num++ {
+		fmt.Fprintln(out, fibinocci(num))
+	}
+}
+func fibinocci(x int) int {
```
Comment (anzdaddy) [inline 01_fibonacci/main.go:16]: Incorrect spelling

**Review by anzdaddy — CHANGES_REQUESTED** [review]

**Comment by anzdaddy** [comment]
- Use the imperative form in the PR title: "create" not "creating"
- Fill in the PR description template.

File: 01_fibonacci/main.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	if x <= 0 {
+		return 0
+	} else if x < 2 {
+		return x
+	}
+	return fibinocci(x-1) + fibinocci(x-2)
+
```
Comment (anzdaddy) [inline 01_fibonacci/main.go:23]: Unnecessary blank line

**Review by anzdaddy — CHANGES_REQUESTED** [review]

**Review by juliaogris — COMMENTED** [review]
-   Click `Update branch` and locally `git pull origin lab1-fibinocci` before adding new commits.
-   Please fix up PR description according to [template](https://github.com/anz-bank/go-training/blob/master/.github/pull_request_template.md)
-   Please edit your commits to follow [recommendations](https://chris.beams.io/posts/git-commit/)
    -   Capitalize the subject line
    -   Use the imperative mood in the subject line
    -   Fold fixup commits into previous ones with `git rebase -i HASH`
I can give a hand with this if you want

**Comment by juliaogris** [comment]
Hi,
There has been  no activity in the last 20 days so I will close this PR.
Feel free to open it again if you have something to add.
Thanks!
Julia

**Pull request closed without merging**